File: 3768-check-if-digits-are-equal-in-string-after-operations-i/3768-check-if-digits-are-equal-in-string-after-operations-i.java
class Solution {
    public boolean hasSameDigits(String s) {

        while(s.length()!=2){
            String ss="";
            for(int i=1;i<s.length();i++){
                int a = s.charAt(i-1)-'0';
                int b = s.charAt(i)-'0';
                int c = (a+b) % 10;
                ss+=String.valueOf(c);
            }
            s=ss;
        }
        
        return s.charAt(0)==s.charAt(1);
    }
}

File: 3768-check-if-digits-are-equal-in-string-after-operations-i/3768-check-if-digits-are-equal-in-string-after-operations-i.cpp
class Solution {
public:
    bool hasSameDigits(string s) {
        
    
        while(s.length()!=2){
            string ss="";
            for(int i=1;i<s.length();i++){
                int a = s[i-1] - '0';
                int b = s[i] - '0';
                int res = (a+b)%10;
                ss+=to_string(res);
            }
            s=ss;
        }

        return s[0]==s[1];
    }
};